client-rpc/src/wallet_rpc.rs

```diff
@@ -202,13 +213,32 @@ where
         }
     }
 
-    fn transactions(&self, request: WalletRequest) -> Result<Vec<TransactionChange>> {
+    fn transactions(&self, request: WalletRequest) -> Result<Vec<RowTx>> {
         self.sync()?;
 
-        match self.client.history(&request.name, &request.passphrase) {
-            Ok(transaction_change) => Ok(transaction_change),
-            Err(e) => Err(to_rpc_error(e)),
-        }
+        self.client
+            .history(&request.name, &request.passphrase)
+            .map_err(|e| to_rpc_error(e))
+            .map(|transaction_changes| {
```

Comment:
Should we port this to the serialization method of `TransactionChange`? Then all future APIs with this structure in response will have the parse-friendly serialization.

Reply:
all future api will be the same format with this tx.
you can use the same decoding api afterwards.